app/controllers/v1/books_controller.rb

```diff
@@ -0,0 +1,6 @@
+class V1::BooksController < ApplicationController
+  def index
+    @pagy, books = pagy(Book.all)
```

Comment:
Marrying a framework will bite you down the line one way or the other. Do you know why / options to limit the impact?

Reply:
I don't exactly know what you mean. Can you be a little more specific, please? Is is the single line assignment or is it the way I declared the BooksController class?

Reply:
Let's imagine that your application grows over time. You end up at some point with pagy in all controllers. At some point, pagy is no longer supported / has some issues and we want to avoid using it. What do we do? We need to either modify all controllers or implement a `pagy` method on the application controller.

Reply:
Nothing to do, just wanted to make a point that dependencies should be as much as possible isolated and not propagated through the codebase.

Reply:
Totally right about that one. If more that one controller was needed, I would've definitely move it to application controller :+1: